core/execution/future/market.go

```diff
@@ -3986,5 +4300,131 @@ func (m *Market) GetRiskFactors() *types.RiskFactor {
+// 1. Any funds in the order margin account will be moved to the margin account.
+// 2. At this point trading can continue with the account switched to the cross margining account type.
+// If there are excess funds in the margin account they will be freed at the next margin release cycle.
+func (m *Market) switchMarginMode(ctx context.Context, party string, marginMode types.MarginMode, marginFactor num.Decimal) error {
+	defer m.onTxProcessed()
+	if marginMode == m.getMarginMode(party) && marginFactor.Equal(m.getMarginFactor(party)) {
+		return nil
+	}
+
+	pos, ok := m.position.GetPositionByPartyID(party)
+	if !ok {
+		pos = positions.NewMarketPosition(party)
+	}
+
+	margins, err := m.collateral.GetPartyMargin(pos, m.settlementAsset, m.GetID())
+	if err == collateral.ErrPartyAccountsMissing {
+		_, err = m.collateral.CreatePartyMarginAccount(ctx, party, m.mkt.ID, m.settlementAsset)
+		if err != nil {
+			return err
+		}
+		margins, err = m.collateral.GetPartyMargin(pos, m.settlementAsset, m.GetID())
+		if err != nil {
+			return err
+		}
+	} else if err != nil {
+		return err
+	}
+
+	marketObservable := m.getMarketObservable(nil)
+	if marketObservable == nil {
+		return fmt.Errorf("no market observable price")
+	}
+	increment := m.tradableInstrument.Instrument.Product.GetMarginIncrease(m.timeService.GetTimeNow().UnixNano())
+	var auctionPrice *num.Uint
+	if m.as.InAuction() {
+		auctionPrice = marketObservable
+		markPrice := m.getCurrentMarkPrice()
+		if markPrice != nil && marketObservable.LT(markPrice) {
+			auctionPrice = markPrice
+		}
+	}
+	// switching to isolated or changing the margin factor
+	if marginMode == types.MarginModeIsolatedMargin {
+		risk, err := m.risk.SwitchToIsolatedMargin(ctx, margins, marketObservable, increment, m.matching.GetOrdersPerParty(party), marginFactor, auctionPrice)
+		if err != nil {
+			return err
+		}
+		// ensure we have an order margiin account set up
+		m.collateral.GetOrCreatePartyOrderMarginAccount(ctx, party, m.mkt.ID, m.settlementAsset)
+		if len(risk) > 0 {
+			for _, r := range risk {
+				err = m.transferMargins(ctx, []events.Risk{r}, nil)
+				if err != nil {
+					return err
+				}
+			}
+		}
+		m.partyMarginFactor[party] = marginFactor
+		// cancel pegged orders
+		ordersAndParkedPegged := append(m.matching.GetOrdersPerParty(party), m.getPartyParkedPeggedOrders(party)...)
+		for _, o := range ordersAndParkedPegged {
+			if o.PeggedOrder != nil {
+				m.cancelOrder(ctx, o.Party, o.ID)
+			}
+		}
+		return nil
+	} else {
+		// switching from isolated margin to cross margin
+		// 1. Any funds in the order margin account will be moved to the margin account.
+		// 2. At this point trading can continue with the account switched to the cross margining account type. If there are excess funds in the margin account they will be freed at the next margin release cycle.
+		risk := m.risk.SwitchFromIsolatedMargin(ctx, margins, marketObservable, increment)
+		if err != nil {
```

Comment:
I don't know what error this is checking, because `SwitchFromIsolatedMargin` isn't returning one.

Reply:
it used to though...